bgpd/bgp_io.c

```diff
@@ -440,8 +439,8 @@ static uint16_t bgp_write(struct peer *peer)
 
 done : {
 	/* Update last_update if UPDATEs were written. */
-	uo = atomic_load_explicit(&peer->update_out, memory_order_relaxed);
-	if (uo > oc)
+	/* Note: that these are only updated at end, not per obuf */
```

Comment:
I don't understand what is meant by this comment, can you clarify?

Reply:
the _out counters are updated in a while loop, and last_update and last_write are only updated after the while obuf processing loop exists.  this means there is a time window where _out is updated but the write times are not.  I don't think this is a significant window, but it's something to note, particularity if / when more io threads are added.

Reply:
Perhaps a comment along the lines of
`/* Note there is a short window where these are out of sync with the tx counters */`
would be more explanatory. The "per obuf" part in particular still makes no sense to me.

Reply:
see new comment.